frontend/lib/resolution/resolution-queries.cpp

```diff
@@ -7481,7 +7481,19 @@ static const types::RuntimeType* const& getRuntimeTypeQuery(Context* context, co
 }
 
 const types::RuntimeType* getRuntimeType(Context* context, const types::CompositeType* ct) {
-  return getRuntimeTypeQuery(context, ct);
+  const DomainType* dom = nullptr;
+  if (auto array = ct->toArrayType()) {
+    dom = array->domainType().type()->toDomainType();
```

Comment:
is `array->domainType()` always non-null? else, there's a segfault in here.

Reply:
I believe we'll always at least make a generic domain type for an array.